### PR TITLE
feat: support constant case

### LINKED ISSUE #309: Support for `constantCase`
nice to have:

```typescript
constantCase("hello world")
// output: HELLO_WORLD
```

`es-toolkit` already supports string *case.

### PR DESCRIPTION
**Pull request by gweesin**
Closed #309

### CODE REVIEW

**Comment by gweesin** [comment]
the ci test error seems caused by the environment:

image

File: docs/zh_hans/reference/string/constantCase.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# constantCase
+
+Converts a string to constant case.
+
+Constant case is a naming convention where each word is written in uppercase letters and separated by an underscore (\_). Such as `CONSTANT_CASE`.
```
Comment (raon0211) [inline docs/zh_hans/reference/string/constantCase.md:5]:
```suggestion
Constant case is a naming convention where each word is written in uppercase letters and separated by an underscore (`_`). For example, `CONSTANT_CASE`.
```

File: src/string/constantCase.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { getWords } from './_internal/getWords.ts';
+
+/**
+ * Converts a string to constant case.
+ *
+ * Snake case is the naming convention in which each word is written in lowercase and separated by an underscore (_) character.
```
Comment (raon0211) [inline src/string/constantCase.ts:6]:
```suggestion
 * Constant case is a naming convention where each word is written in uppercase letters and separated by an underscore (`_`). For example, `CONSTANT_CASE`.
```

**Review by raon0211 — APPROVED** [review]
Thanks!

**Pull request merged**